src/index.js

```diff
@@ -17,6 +17,28 @@ export function t(strings, ...exprs) {
     return strings;
 }
 
+const separator = /(\${\s*\d+\s*})/g;
+const slotIdRegexp = /\${\s*(\d+)\s*}/;
+
+export function jt(strings, ...exprs) {
+    if (strings && strings.reduce) {
+        const id = getMsgid(strings, exprs);
+        const transObj = findTransObj(currentLocale, id);
+        if (!transObj) return [id];
+
+        // splits string & capturing group into tokens
+        //
+        const translatedTokens = transObj.msgstr[0].split(separator);
+
+        return translatedTokens.map((token) => {
```

Comment:
Translated strings may sort expressions differently from `msgid`s. Therefore the index inside `${ }` is important.

For example, an message with `msgid` like
`You have viewed ${0} product in ${1} week`

Maybe translated to this in Chinese:
`你在 ${1} 週內已瀏覽 ${0} 件商品`

As far as I know, Chinese puts place & time in front of verbs, and Japanese puts objects in front of transitive verbs. This is tested by the test case `should handle the case that variable positions are inconsistent`.

In `t` we don't have this issue because `string.replace` was used, but when constructing arrays, it is necessary to map the variables to items in `exprs`.